File: conversions/binary-to-decimal/toDecimal.js
/**
 * Binary to decimal conversion
 * @param {Array} bin binary array
 * @returns {Number}
 */
function toDecimal(bin) {
  const size = bin.length;
  let dec = 0;
  for (let i = 0; i < size; i++) {
    dec += bin[i] << (size - i - 1);
  }
  return dec;
}

function randomBinary(size) {
  return [1, ...[...new Array(Math.floor(Math.random() * size)).keys()].map(() => (Math.random() > 0.5 ? 1 : 0))];
}

/**
 * Helper function to log binary array as string
 * @param {Array} seq
 * @returns {String}
 */
function asString(seq) {
  let str = "";
  seq.forEach((x) => {
    str += String(x);
  });
  return str;
}

function main() {
  console.log();
  for (let i = 0; i < 10; i++) {
    const bin = randomBinary(6);
    const dec = toDecimal(bin);
    console.log(`binary: ${asString(bin)}\t--> decimal: ${dec}`);
  }
}

main();

File: conversions/binary-to-decimal/to_decimal.py
from random import randint


def to_decimal(bin: list) -> int:
    """ convert binary array to decimal """
    size = len(bin)
    return sum([x << (size - i - 1) for i, x in enumerate(bin)])


def random_binary():
    return [1] + [randint(0, 1) for _ in range(randint(0, 6))]


def as_str(seq):
    y = ''
    for x in seq:
        y += str(x)
    return y


if __name__ == '__main__':
    print()
    for _ in range(10):
        bin = random_binary()
        dec = to_decimal(bin)
        print(f'binary: {as_str(bin)}\t-->\tdecimal: {dec}')
    print()
